resqpy/well/_wellbore_marker.py

```diff
@@ -0,0 +1,167 @@
+             e.g. "fault", "geobody", "horizon ", "gas/oil/water down to", "gas/oil/water up to",
+              "free water contact", "gas oil contact", "gas water contact", "water oil contact", "seal"
+           interpretation_uuid (uuid.UUID or string, optional): uuid of the boundary feature Interpretation
+              organizational object that the marker refers to.
+              note: it is highly recommended that a related boundary feature interpretation is provided
+           title (str, optional): the citation title to use for a new wellbore marker
+              ignored if uuid is not None;
+           originator (str, optional): the name of the person creating the wellbore marker, defaults to login id;
+              ignored if uuid is not None
+           extra_metadata (dict, optional): string key, value pairs to add as extra metadata for the wellbore marker;
+              ignored if uuid is not None
+
+        returns:
+           the newly created wellbore marker object
+        """
+        # verify that marker type is valid
+        if marker_type is not None:
+            assert marker_type in ([
+                "fault", "geobody", "horizon", "gas down to", "oil down to", "water down to", "gas up to", "oil up to",
+                "water up to", "free water contact", "gas oil contact", "gas water contact", "water oil contact", "seal"
+            ]), "invalid marker type specified"
+
+        self.model = parent_model
+        self.wellbore_frame = parent_frame
+        self.marker_index = marker_index
+        self.uuid = None
+        self.marker_type = marker_type
+        self.interpretation_uuid = interpretation_uuid
+        self.title = title
+        self.originator = originator
+        self.extra_metadata = extra_metadata
+        if marker_node is not None:
+            self._load_from_xml(marker_node = marker_node)
+        if self.uuid is None:
+            self.uuid = bu.new_uuid()
+        assert self.uuid is not None
+
+    def create_xml(self, parent_node, title = 'wellbore marker'):
+        """Creates the xml tree for this wellbore marker.
+
+        arguments:
+           parent_node (xml node): the root node of the WellboreMarkerFrame object to which the newly created node will be appended
+           title (string, optional): the citation title of the newly created node
+             note: if not None, self.title will be used instead of "wellbore marker"
+
+        returns:
+           the newly created xml node
+        """
+
+        assert self.uuid is not None
+        wbm_node = self.model.new_obj_node('WellboreMarker', is_top_lvl_obj = False)
+        wbm_node.set('uuid', str(self.uuid))
+
+        # Citation block
+        if self.title is not None:
+            title = self.title
+        citation = self.model.create_citation(root = wbm_node, title = title, originator = self.originator)
+
+        # Add sub-elements to root node
+        boundary_feature_dict = {
```

Comment:
I generally prefer to put constant data like this dict outside the method, either as a class variable or a global variable outside the class.